channel/pools.go

```diff
@@ -1051,10 +1075,50 @@ func packetShowDrop(spawnType byte, drop fieldDrop) mpacket.Packet {
 	return p
 }
 
-func packetRemoveDrop(instant bool, dropID int32) mpacket.Packet {
+func packetRemoveDrop(dropType int8, dropID int32, lootedBy int32) mpacket.Packet {
 	p := mpacket.CreateWithOpcode(opcode.SendChannelDropExitMap)
-	p.WriteBool(instant) // 0 - fade away, 1 - instant, 2,3,5 - player id? , 4 - int16
+	p.WriteInt8(dropType) // 0 - fade away, 1 - instant, 2 - loot by user, 3 - loot by mob
 	p.WriteInt32(dropID)
+	p.WriteInt32(lootedBy)
+
+	return p
+}
+
+func packetPickupNotice(itemID int32, amount int16, isMesos bool, isEquip bool) mpacket.Packet {
```

Comment:
From what I can tell the packet structure is as follows:
```
func packetPickupNotice(itemID int32, amount int16, isMesos bool, isEquip bool, cafeBonus int16) mpacket.Packet {
	p := mpacket.CreateWithOpcode(opcode.SendChannelInfoMessage)
	p.WriteInt8(0) //??

	p.WriteBool(isMesos)

	if isMesos {
		p.WriteInt32(int32(amount))
		p.WriteInt16(cafeBonus)
	} else {
		p.WriteInt32(itemID)

		if !isEquip {
			p.WriteInt16(amount)
		}

		p.WriteInt32(0)
	}

	return p
}
```

Reply:
That worked perfectly, ah when I tried before I didn't cast the amount as int32, oops. Thank you for it. I made the changes.